pennylane/interfaces/jax.py

```diff
@@ -373,3 +374,201 @@ def wrapped_exec_bwd(params, g):
+            (classical) computational overhead during the backwards pass.
+
+    Returns:
+        list[list[float]]: A nested list of tape results. Each element in
+        the returned list corresponds in order to the provided tapes.
+    """
+    # Set the trainable parameters
+    for tape in tapes:
+        params = tape.get_parameters(trainable_only=False)
+        tape.trainable_params = qml.math.get_trainable_indices(params)
+
+    parameters = tuple(list(t.get_parameters()) for t in tapes)
+
+    if gradient_fn is None:
+        # PennyLane forward execution
+        return _execute_fwd_new(
+            parameters,
+            tapes=tapes,
+            execute_fn=execute_fn,
+            gradient_kwargs=gradient_kwargs,
+            _n=_n,
+        )
+
+    # PennyLane backward execution
+    return _execute_bwd_new(
+        parameters,
+        tapes=tapes,
+        execute_fn=execute_fn,
+        gradient_fn=gradient_fn,
+        gradient_kwargs=gradient_kwargs,
+        _n=_n,
+        max_diff=max_diff,
+    )
+
+
+def _execute_fwd_new(
+    params,
+    tapes,
+    execute_fn,
+    gradient_kwargs,
+    _n=1,
+):
+    """The auxiliary execute function for cases when the user requested
+    jacobians to be computed in forward mode (e.g. adjoint) or when no gradient function was
+    provided. This function is does not allow multiple derivatives."""
+
+    # pylint: disable=unused-variable
+    @jax.custom_jvp
+    def execute_wrapper(params):
+        new_tapes = [_copy_tape(t, a) for t, a in zip(tapes, params)]
+
+        with qml.tape.Unwrap(*new_tapes):
+            res, jacs = execute_fn(new_tapes, **gradient_kwargs)
+
+        res = _to_jax(res)
+
+        return res, jacs
+
+    @execute_wrapper.defjvp
+    def execute_wrapper_jvp(primal, tangents):
+        res, jacs = execute_wrapper(primal[0])
+        multi_measurements = [len(tape.measurements) > 1 for tape in tapes]
+
+        jvps = _compute_jvps(jacs, tangents[0], multi_measurements)
+        return res, jvps
+
+    res = execute_wrapper(params)
+
+    tracing = any(isinstance(r, jax.interpreters.ad.JVPTracer) for r in res)
+
+    # When there are no tracers (not differentiating), we have the result of
+    # the forward pass and the jacobian, but only need the result of the
+    # forward pass
+    if len(res) == 2 and not tracing:
+        res = res[0]
+
+    return res
+
+
+def _compute_jvps(jacs, tangents, multi_measurements):
+    """Compute the jvps of multiple tapes, directly for a Jacobian and tangents."""
+    jvps = []
+    for i, multi in enumerate(multi_measurements):
+        if multi:
+            jvps.append(qml.gradients.compute_jvp_multi(tangents[i], jacs[i]))
+        else:
+            jvps.append(qml.gradients.compute_jvp_single(tangents[i], jacs[i]))
+    return jvps
+
+
+def _execute_bwd_new(
+    params,
+    tapes,
+    execute_fn,
+    gradient_fn,
+    gradient_kwargs,
+    _n=1,
+    max_diff=1,
```

Comment:
It's good to keep default values consistent even if this is private.
```suggestion
    max_diff=2,
```